### PR TITLE
Add filename to importers and log it where possible

### PR DESCRIPTION
**Pull request by kaklakariada**
See ticket #37

### CODE REVIEW

File: src/main/java/openfasttrack/importer/markdown/MarkdownImporter.java [hunk 1]
```diff
@@ -66,7 +68,8 @@ public void runImport()
         }
         catch (final IOException exception)
         {
-            LOG.warning("IO exception after line " + lineNumber);
+            LOG.warning(
+                    "IO exception after line " + lineNumber + " of file '" + this.fileName + "'");
```
Comment (redcatbear) [inline src/main/java/openfasttrack/importer/markdown/MarkdownImporter.java]: I would prefer all input related warnings to be in a uniform, parseable format (e.g. Java stack-trace-style). in the best case that allows editors like Eclipse to jump to the file in question.

File: src/main/java/openfasttrack/importer/specobject/ImportHelper.java [hunk 2]
```diff
@@ -102,7 +106,8 @@ private void foundStartElement(final StartElement element) throws XMLStreamExcep
         switch (element.getName().getLocalPart())
         {
         case SPECDOCUMENT_ROOT_ELEMENT_NAME:
-            LOG.finest("Found XML root element '" + SPECDOCUMENT_ROOT_ELEMENT_NAME + "': ignore.");
+            LOG.finest("Found XML root element '" + SPECDOCUMENT_ROOT_ELEMENT_NAME + "' in file '"
```
Comment (redcatbear) [inline src/main/java/openfasttrack/importer/specobject/ImportHelper.java]: Same comment as above: Uniform and parseable.

File: src/main/java/openfasttrack/importer/specobject/ImportHelper.java [hunk 3]
```diff
@@ -111,28 +116,32 @@ private void foundStartElement(final StartElement element) throws XMLStreamExcep
...
                 LOG.finest(() -> "Found XML element '" + SPECOBJECTS_ELEMENT_NAME
-                        + "' with default doctype '" + this.defaultDoctype + "'.");
+                        + "' with default doctype '" + this.defaultDoctype + "' in file '"
+                        + this.fileName + "'.");
             }
             else
             {
                 throw new ImporterException("Element " + element + " does not have an attribute '"
-                        + doctypeAttributeName + "'.");
+                        + doctypeAttributeName + "' in file '" + this.fileName + "'.");
```
Comment (redcatbear) [inline src/main/java/openfasttrack/importer/specobject/ImportHelper.java]: dito.

**Comment by kaklakariada** [comment]
Regarding the filename for stdin: We could use a dummy filename like 'stdin'.

File: src/main/java/openfasttrack/importer/markdown/MarkdownImporter.java (hunk 1 above)
Comment (kaklakariada) [inline src/main/java/openfasttrack/importer/markdown/MarkdownImporter.java]: Using format filename:line

File: src/main/java/openfasttrack/importer/specobject/ImportHelper.java (hunk 2 above)
Comment (kaklakariada) [inline src/main/java/openfasttrack/importer/specobject/ImportHelper.java]: Using format filename:line

File: src/main/java/openfasttrack/importer/specobject/ImportHelper.java (hunk 3 above)
Comment (kaklakariada) [inline src/main/java/openfasttrack/importer/specobject/ImportHelper.java]: Using format filename:line

File: src/main/java/openfasttrack/importer/specobject/ImportHelper.java [hunk 4]
```diff
@@ -87,7 +90,9 @@ private boolean isXmlRootWhitelisted(final StartElement currentElement)
             this.validRootElementFound = true;
             return false;
         }
-        LOG.warning(() -> "Found unknown XML root element '" + currentElement + "': skip file.");
+        LOG.warning(
+                () -> "Found unknown XML root element '" + currentElement + "' at " + this.fileName
+                        + ":" + currentElement.getLocation().getLineNumber() + ": skip file.");
```
Comment (redcatbear) [inline src/main/java/openfasttrack/importer/specobject/ImportHelper.java]: Code duplication. Please extract. Otherwise fine.

Comment (kaklakariada) [inline src/main/java/openfasttrack/importer/specobject/ImportHelper.java]: Duplication removed.

**Pull request merged**